### PR TITLE
feat: add URI handlers

### LINKED ISSUE #73: URI methods
Add methods to deconstruct URI's that are retrieved with LDflex; specifically `.prefix`, `.namespace` and `.fragment`

A rudimentary way that I am currently achieving this in the branch that I am working on is  below - though I would move the API call to prefix.cc if I created a PR and only return the prefix if we can establish it from the context (or it can be somehow obtained through the engine that is being used)

```ts

function subjectToComponentsHandler(component: 'namespace' | 'fragment' | 'prefix') {

  return handler(async ({ subject, prefixes = {} }: Data, path): Promise<string | undefined> => {
    if (subject?.termType === 'NamedNode') {
      if (component === 'namespace') {
        return /^[^]*[#\/]/.exec(subject.value)?.[0]
      } else if (component === 'fragment') {
        return /(?![\/#])[^\/#]*$/.exec(subject.value)?.[0]
      } else if (component === 'prefix') {
        const ns = /^[^]*[#\/]/.exec(subject.value)?.[0]
        const pref = ns ? prefixes[ns] : undefined
        try {
          // TODO: Get prefixes from the engines first (if possible)
          const prefix =  pref ?? /[a-z]*$/i.exec((await fetch(`http://prefix.cc/reverse?uri=${ns}`)).url)?.[0] ?? undefined;
          ns && prefix &&  (prefixes[ns] = prefix);
          return prefix
        } catch {
          return undefined
        }
      }
    } else
      return undefined
  })

}
```

### PR DESCRIPTION
**Pull request by jeswr**
Closes #73 .

Adds `.prefix`, `.namespace` and `.fragment` handlers, which have the following behavior on a subject that is a NamedNode (the handlers return `undefined` on other termTypes)

```js
ruben.namespace // 'https://ruben.verborgh.org/profile/#'
ruben.fragment // 'me'
await ruben.prefix // 'rbn'
```

I'm proposing this addition now as it is of use in a project that I am working on - but won't be offended if you do not think it is appropriate to add it here.

### CODE REVIEW

**Comment by jeswr** [comment]
@RubenVerborgh /bump

File: src/URIHandler.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { handler } from './handlerUtil';
+
+const NAMESPACE = /^[^]*[#/]/;
```
Comment (RubenVerborgh) [inline src/URIHandler.js:3]:
```suggestion
const NAMESPACE = /^[^#]*[#/]/;
```

File: src/URIHandler.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+const FRAGMENT = /(?![/#])[^/#]*$/;
```
Comment (RubenVerborgh) [inline src/URIHandler.js:4]: This one is not correct. (`http://abc/def#abc/def`)
Perhaps best to use `NAMESPACE` to strip off the beginning; or change it to have 2 capturing groups.

**Review by RubenVerborgh — APPROVED** [review]
Good to go if regexes are adjusted.

**Pull request merged**